os/fs/smartfs/smartfs_smart.c

```diff
@@ -637,58 +639,62 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 
 	smartfs_semtake(fs);
 
-	/* Test the permissions.  Only allow write if the file was opened with
-	 * write flags.
-	 */
+	/* Test the permissions. Only allow write if the file was opened with write flags */
 
 	if ((sf->oflags & O_WROK) == 0) {
 		ret = -EACCES;
 		goto errout_with_semaphore;
 	}
 
-	/* First test if we are overwriting an existing location or writing to
-	 * a new one. */
-
 	header = (struct smartfs_chain_header_s *)fs->fs_rwbuffer;
 	byteswritten = 0;
+	size = sizeof(struct smartfs_chain_header_s);
```

Comment:
it never changed, you'd better initialize it directly in line 612.

Reply:
Ohkay